lib/sys/system.flow

```diff
@@ -21,4 +21,13 @@ export {
 
 	// start a process detached from the application process
 	native startDetachedProcess : (command : string, args : [string], currentWorkingDirectory : string) -> bool = Native.startDetachedProcess;
+
+	// Read n bytes from stdin input
+	native readBytes : io (n : int) -> string = Native.readBytes;
```

Comment:
What encoding is used? How are bytes represented in the string? Is it 8-bits in the lower parts, is it interpreted as UTF-8, what is the encoding?

Reply:
These functions operate on a byte representation of strings - read n bytes, and compare a byte string stream with a pattern byte-wise. The resulting string in java is formed from a byte array as:
`byte[] bytes = new byte[line.size()]; .... return new String(bytes);`
So it uses a default encoding.
Maybe here I should rather use `return new String(bytes, "UTF-8");` - I'm not sure.

Reply:
The important thing is to document what it does.